synapse/handlers/message.py

```diff
@@ -1416,9 +1416,11 @@ async def _rebuild_event_after_third_party_rules(
 
         # the event type hasn't changed, so there's no point in re-calculating the
         # auth events.
+        prev_event_ids = await self.store.get_prev_events_for_room(builder.room_id)
+
         event = await builder.build(
-            prev_event_ids=original_event.prev_event_ids(),
-            auth_event_ids=original_event.auth_event_ids(),
+            prev_event_ids=prev_event_ids,
+            auth_event_ids=None,
```

Comment:
why is this changing? it seems to now do the exact opposite to what the comment says?

Reply:
Ugh, looks like I forgot to finish that bit up. I also wanted to PR it to mainline so it doesn't end up producing another divergence, I've done that here: https://github.com/matrix-org/synapse/pull/10316

Reply:
the comment still needs updating for this branch.